### PR TITLE
Adding write control to IO file backends

### PR DESCRIPTION
**Pull request by NickGeneva**
# Earth2Studio Pull Request

## Description

Doing some updates to the file IOs to accept kwargs for the storage backends to give slightly better control over storage file behavior.

Closes: https://github.com/NVIDIA/earth2studio/issues/105

## Checklist

- [x] I am familiar with the [Contributing Guidelines](https://github.com/NVIDIA/earth2studio/blob/main/CONTRIBUTING.md).
- [x] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.
- [x] The [CHANGELOG.md](https://github.com/NVIDIA/earth2studio/blob/main/CHANGELOG.md) is up to date with these changes.
- [x] An [issue](https://github.com/NVIDIA/earth2studio/issues) is linked to this pull request.

### CODE REVIEW

File: earth2studio/io/netcdf4.py [hunk 1]
```diff
@@ -31,31 +32,25 @@ class NetCDF4Backend:
...
-        File name to provide for creating the netcdf4 store.
-    diskless : bool, optional
-        Whether to store the Dataset in memory buffer
-        Default value is False.
-    persist : bool, optional
-        Whether to save in-memory diskless buffer to disk upon close.
-        Default value is False.
+    file_name : str
+        _description_
+    nc_kwargs : dict[str, Any], optional
```
Comment (dallasfoster) [inline earth2studio/io/netcdf4.py]: Can we make this `backend_kwargs` to be comparable with `xarray`?

File: earth2studio/io/zarr.py [hunk 2]
```diff
@@ -32,23 +32,32 @@ class ZarrBackend:
...
+    Note
+    ----
+    For keyword argument options see: https://zarr.readthedocs.io/en/stable/api/hierarchy.html
     """
 
     def __init__(
         self,
         file_name: str = None,
         chunks: dict[str, int] = {},
+        zarr_kwargs: dict[str, Any] = {"overwrite": False},
```
Comment (dallasfoster) [inline earth2studio/io/zarr.py]: Similar, can we set this to be `backend_kwargs`.

File: earth2studio/io/netcdf4.py (hunk 1 above)
Comment (NickGeneva) [reply earth2studio/io/netcdf4.py]: Good idea, updated :+1:

File: earth2studio/io/zarr.py (hunk 2 above)
Comment (NickGeneva) [reply earth2studio/io/zarr.py]: Updated as well!

**Review by dallasfoster — APPROVED** [review]

**Pull request merged**